### ISSUE TITLE
AMI Creation Feature

### ISSUE THREAD

**Issue by tanmayIITR** [cloud-custodian/cloud-custodian#5122]
Hi! Can we have an ami creation feature integrated with the ec2 actions? I needed to make ami before terminating the non compliant instances. I used a lambda function to do it and trigger it via webhook in the .yml policy.

**Comment by kapilt** [comment]
first you can invoke a lambda directly via invoke-lambda.. a web hook would be pretty suspect from a security perspective.

re create-ami instance action, that sounds good, this project is community lead and driven so that will get done faster if there are contributions for it. most of the existing aws maintainers are going to be catching up on reinvent announcements for a bit.

**Comment by kapilt** [comment]
also this is a duplicate of #4420

**Comment by kapilt** [comment]
per my comments on the other issue, its worth looking at just using backup manager for most of the use cases of creating an ami for an instance as a backup.

**Comment by tanmayIITR** [comment]
Yes, I looked at invoke-lambda action but wasn't sure how to use it with a JMESPath Query, hence used a webhook and API-Gateway to trigger the lambda.
Can you guide me how I can achieve this using invoke-lambda action?

**Comment by tanmayIITR** [comment]
I've figured it out using invoke-lambda..Thanks!
